packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,432 @@
+            || _addr == Predeploys.SEQUENCER_FEE_WALLET || _addr == Predeploys.OPTIMISM_MINTABLE_ERC20_FACTORY
+            || _addr == Predeploys.OPTIMISM_MINTABLE_ERC721_FACTORY || _addr == Predeploys.L1_BLOCK_ATTRIBUTES
+            || _addr == Predeploys.GAS_PRICE_ORACLE || _addr == Predeploys.DEPLOYER_WHITELIST || _addr == Predeploys.WETH9
+            || _addr == Predeploys.L1_BLOCK_NUMBER || _addr == Predeploys.LEGACY_MESSAGE_PASSER
+            || _addr == Predeploys.PROXY_ADMIN || _addr == Predeploys.BASE_FEE_VAULT || _addr == Predeploys.L1_FEE_VAULT
+            || _addr == Predeploys.GOVERNANCE_TOKEN || _addr == Predeploys.SCHEMA_REGISTRY || _addr == Predeploys.EAS;
+    }
+
+    /// @dev Function to compute the expected address of the predeploy implementation
+    ///      in the genesis state.
+    function _predeployToCodeNamespace(address _addr) internal pure returns (address) {
+        return address(
+            uint160(uint256(uint160(_addr)) & 0xffff | uint256(uint160(0xc0D3C0d3C0d3C0D3c0d3C0d3c0D3C0d3c0d30000)))
+        );
+    }
+
+    function _setImplementationCode(address _predeployAddr, string memory _predeployName) internal returns (address) {
+        address impl = _predeployToCodeNamespace(_predeployAddr);
+        console.log("Setting %s implementation at: %s", _predeployName, impl);
+        vm.etch(impl, vm.getDeployedCode(string.concat(_predeployName, ".sol:", _predeployName)));
+
+        _verifyProxyImplementationAddress(_predeployAddr, impl);
+
+        return impl;
+    }
+
+    /// @dev Function to verify the expected implementation address is set for the respective proxy.
+    function _verifyProxyImplementationAddress(address _proxy, address _impl) internal view {
+        require(
+            EIP1967Helper.getImplementation(_proxy) == _impl,
+            "Expected different address at Proxys PROXY_IMPLEMENTATION_ADDRESS storage slot"
+        );
+    }
+
+    /// @dev Function to verify that a contract was initialized, and can't be reinitialized.
+    /// @notice There isn't a good way to know if the resulting revering is due to abi mismatch
+    ///         or because it's already been initialized
+    function _verifyCantReinitialize(address _contract, address _arg) internal {
+        vm.expectRevert("Initializable: contract is already initialized");
+        IInitializable(_contract).initialize(_arg);
+    }
+
+    /// @dev Helper function to sort the genesis alloc numerically by address.
+    function _sortJsonByKeys(string memory _path) internal {
+        string[] memory commands = new string[](3);
+        commands[0] = "bash";
+        commands[1] = "-c";
+        commands[2] = string.concat("cat <<< $(jq -S '.' ", _path, ") > ", _path);
+        vm.ffi(commands);
+    }
+
+    function _fundDevAccounts() internal {
+        for (uint256 i; i < devAccounts.length; i++) {
+            console.log("Funding dev account %s with %s ETH", devAccounts[i], DEV_ACCOUNT_FUND_AMT / 1e18);
```

Comment:
Why the division?

Reply:
ah i see its a log

Reply:
To print `10000` instead of 10,000 in wei